File: app/support/bulkImport/SupervisorState.java
package support.bulkImport;

import org.joda.time.DateTime;

/**
 * Created with IntelliJ IDEA.
 * User: matthijs
 * Date: 7/11/13
 * Time: 8:59 PM
 * To change this template use File | Settings | File Templates.
 */
public class SupervisorState {

    private int succesCount = 0;
    private int failureCount = 0;
    private int timeOutcount = 0;

    private DateTime startTime = new DateTime();

    private int workers = 0;
    private int activeWorkers = 0;
    private int payloadCount = 0;

    private ImportSupervisorActor.Status status = ImportSupervisorActor.Status.STOPPED;

    private long transformerId;
    private String currentFile;
    private long nrOfLines;



    public long getNrOfLines() {
        return nrOfLines;
    }

    public String getCurrentFile() {
        return currentFile;
    }

    public void setCurrentFileSpecs(String currentFile, long nrOfLines) {
        this.currentFile = currentFile;
        this.nrOfLines = nrOfLines;
    }


    public long getTransformerId() {
        return transformerId;
    }

    public void setTransformerId(long transformerId) {
        this.transformerId = transformerId;
    }

    public int getSuccesCount() {
        return succesCount;
    }

    public void incrementSuccesCount() {
        this.succesCount++;
        resetTimeOutCount();
    }

    public int getFailureCount() {
        return failureCount;
    }

    public void incrementFailureCount() {
        this.failureCount++;
        resetTimeOutCount();
    }

    public int getTimeOutcount() {
        return timeOutcount;
    }

    public void resetTimeOutCount() {
        this.timeOutcount=0;
    }

    public void incrementTimeOutCount() {
        this.timeOutcount++;
    }

    public DateTime getStartTime() {
        return startTime;
    }

    public void setStartTime(DateTime startTime) {
        this.startTime = startTime;
    }

    public void setWorkers(int count) {
        this.workers = count;
    }

    public int getWorkers() {
        return workers;
    }


    public void incrementActiveWorkers() {
        activeWorkers++;
    }

    public void decrementActiveWorkers() {
        activeWorkers--;
    }


    public int getActiveWorkers() {
        return activeWorkers;
    }

    public int getPayloadCount() {
        return payloadCount;
    }

    public void incrementPayloadCount() {
        this.payloadCount++;
    }

    public synchronized ImportSupervisorActor.Status getStatus() {
        return status;
    }

    public synchronized void setStatus(ImportSupervisorActor.Status status) {
        this.status = status;
    }


}

File: app/support/bulkImport/SupervisorState.scala
package support.bulkImport

import org.joda.time.DateTime

/**
 * Created with IntelliJ IDEA.
 * User: matthijs
 * Date: 7/11/13
 * Time: 8:59 PM
 * To change this template use File | Settings | File Templates.
 */
class SupervisorState {
  def getNrOfLines: Long = {
    nrOfLines
  }

  def getCurrentFile: String = {
    currentFile
  }

  def setCurrentFileSpecs(currentFile: String, nrOfLines: Long) {
    this.currentFile = currentFile
    this.nrOfLines = nrOfLines
  }

  def getTransformerName: String = {
    transformerName
  }

  def setTransformerName(transformerName: String) {
    this.transformerName = transformerName
  }

  def getTransformerId: Long = {
    transformerId
  }

  def setTransformerId(transformerId: Long) {
    this.transformerId = transformerId
  }

  def getSuccesCount: Int = {
    succesCount
  }

  def incrementSuccesCount {
    this.succesCount += 1
    resetTimeOutCount
  }

  def getFailureCount: Int = {
    failureCount
  }

  def incrementFailureCount {
    this.failureCount += 1
    resetTimeOutCount
  }

  def getTimeOutcount: Int = {
    timeOutcount
  }

  def resetTimeOutCount {
    timeOutcount = 0
  }

  def incrementTimeOutCount {
    this.timeOutcount += 1
  }

  def getStartTime: DateTime = {
    startTime
  }

  def setStartTime(startTime: DateTime) {
    this.startTime = startTime
  }

  def getStopTime: DateTime = {
    stopTime
  }

  def setStopTime(stopTime: DateTime) {
    this.stopTime = stopTime
  }

  def setWorkers(count: Int) {
    this.workers = count
  }

  def getWorkers: Int = {
    workers
  }

  def incrementActiveWorkers {
    activeWorkers += 1
  }

  def decrementActiveWorkers {
    activeWorkers -= 1
  }

  def getActiveWorkers: Int = {
    activeWorkers
  }

  def getPayloadCount: Int = {
    payloadCount
  }

  def incrementPayloadCount {
    this.payloadCount += 1
  }

  def getStatus: ImportSupervisorActor.Status = {
    status
  }

  def setStatus(status: ImportSupervisorActor.Status) {
    this.status = status
  }

  private var succesCount: Int = 0
  private var failureCount: Int = 0
  private var timeOutcount: Int = 0
  private var startTime: DateTime = new DateTime
  private var stopTime: DateTime = new DateTime
  private var workers: Int = 0
  private var activeWorkers: Int = 0
  private var payloadCount: Int = 0
  private var status: ImportSupervisorActor.Status = ImportSupervisorActor.Status.STOPPED
  private var transformerId: Long = 0L
  private var transformerName: String = null
  private var currentFile: String = null
  private var nrOfLines: Long = 0L
}